src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,99 @@
+package duke;
+
+import javafx.application.Application;
+import javafx.scene.Scene;
+import javafx.scene.control.Button;
+import javafx.scene.control.Label;
+import javafx.scene.control.ScrollPane;
+import javafx.scene.control.TextField;
+import javafx.scene.image.Image;
+import javafx.scene.image.ImageView;
+import javafx.scene.layout.AnchorPane;
+import javafx.scene.layout.Region;
+import javafx.scene.layout.VBox;
+import javafx.stage.Stage;
+
+import duke.controllers.MainWindow;
+import duke.command.Command;
+import duke.exception.DukeException;
+import duke.parser.Parser;
+import duke.storage.Storage;
+import duke.ui.Ui;
+import duke.list.FunnyList;
+
+/**
+ * Main class for the Duke application.
+ * This class initializes and manages the Duke application. It handles user interaction,
+ * task processing, and exception handling.
+ */
+public class Duke {
+    private Storage storage;
```

Comment:
Overall I think your code looks good, just some few nits here and there, other than that good job! 😄